app/models/course/condition/achievement.rb

```diff
@@ -3,4 +3,39 @@ class Course::Condition::Achievement < ActiveRecord::Base
   belongs_to :achievement, class_name: Course::Achievement.name
 
   default_scope { includes(:achievement) }
+
+  validates :achievement, presence: true
+  validate :valid_achievement_condition, if: :new_record?
+
+  delegate :title, to: :achievement
+
+  private
+
+  # Given an achievement, returns all its achievement conditions.
+  #
+  # @param [Course::Achievement] achievement The achievement to return achievement conditions for.
+  # @return [Array<Course::Achievement>]
+  def achievement_conditions(achievement)
+    Course::Achievement.joins(
+      "INNER JOIN
+        (SELECT cca.achievement_id
+          FROM course_condition_achievements cca INNER JOIN course_conditions cc
+            ON cc.actable_type = 'Course::Condition::Achievement' AND cc.actable_id = cca.id
+            WHERE cc.conditional_id = #{achievement.id}) ids
+    ON ids.achievement_id = course_achievements.id")
+    # Course::Condition::Achievement.
```

Comment:
Document why this is used. Comments like this would be removed over time. It's always the _why_